skore/src/skore/project/metadata.py

```diff
@@ -37,6 +40,135 @@ class Metadata(DataFrame):
+            itemized_pipeline.append(step.numeric)
+            itemized_pipeline.append(step.datetime)
+        else:
+            itemized_pipeline.append(step)
+        return itemized_pipeline
+
+    @staticmethod
+    def _explore_sk_pipeline(
+        pipeline: Pipeline, itemized_pipeline: list, depth: int = 0, max_depth: int = 5
+    ) -> list:
+        """
+        Explore the pipeline and return a list of estimators or transformers.
+
+        Parameters
+        ----------
+        pipeline : Pipeline
+            The pipeline to explore.
+
+        Returns
+        -------
+        list
+            A list of estimators or transformers in the pipeline.
+        """
+        if depth > max_depth:
+            return itemized_pipeline
+
+        for _, step in pipeline.steps:
+            itemized_pipeline = Metadata._iterate_in_pipe_tree(
+                step, itemized_pipeline, depth=depth
+            )
+        return itemized_pipeline
+
+    @staticmethod
+    def _explore_sk_feature_union(
+        feature_union: FeatureUnion,
+        itemized_pipeline: list,
+        depth: int = 0,
+        max_depth: int = 5,
+    ) -> list:
+        """
+        Explore the feature union and return a list of estimators or transformers.
+
+        Parameters
+        ----------
+        feature_union : FeatureUnion
+            The feature union to explore.
+
+        Returns
+        -------
+        list
+            A list of estimators or transformers in the feature union.
+        """
+        if depth > max_depth:
+            return itemized_pipeline
+
+        for _, transformer in feature_union.transformer_list:
+            itemized_pipeline = Metadata._iterate_in_pipe_tree(
+                transformer, itemized_pipeline, depth=depth
+            )
+        return itemized_pipeline
+
+    @staticmethod
+    def _split_pipeline(pipeline: Pipeline) -> dict[str, list[str]]:
+        """
+        Split the pipeline into a list of estimators or transformers.
+
+        Parameters
+        ----------
+        pipeline : Pipeline
+            The pipeline to split.
+
+        Returns
+        -------
+        dict[str, list[str]]
+            A dict of estimators or transformers class in the pipeline,
+            grouped by their module name.
+        """
+        max_depth = 5
+        itemized_pipeline = Metadata._explore_sk_pipeline(
+            pipeline, [], depth=0, max_depth=max_depth
+        )
+        class_names = list(set([str(type(est)) for est in itemized_pipeline]))
+
+        sklearn_regex = re.compile(r"sklearn\.(\w+)\.(\w+)")
+        skrub_regex = re.compile(r"skrub\.(\w+)\.(\w+)")
+
+        # pipe_steps: dict[str, list[str]] = dict()
+        pipe_steps = defaultdict(list)
+        for estimator_class in class_names:
+            sklearn_match = sklearn_regex.search(estimator_class)
+            skrub_match = skrub_regex.search(estimator_class)
+            if sklearn_match is not None:
```

Comment:
nit: This would be the use-case for the walrus (`:=`) operator:
```suggestion
            if (sklearn_match := sklearn_regex.search(estimator_class)) is not None:
```

Reply:
This is the typical use-case. But I find it more difficult to read.